build-tools/automation/yaml-templates/kill-processes.yaml

```diff
@@ -1,12 +1,16 @@
 steps:
 - powershell: |
-    $ErrorActionPreference = 'Continue'
+    $ErrorActionPreference = 'SilentlyContinue'
     Get-Process
-    Stop-Process -Name nunit3-console
-    Stop-Process -Name xabuild
-    Stop-Process -Name MSBuild
-    Stop-Process -Name adb
-    $LastExitCode=0
+    Write-Host "Attempting to kill stray processes..."
+    Stop-Process -Name nunit3-console -ErrorVariable nunitError -PassThru
+    if ($nunitError) { Write-Host "A process named 'nunit3-console' could not be found or stopped."}
+    Stop-Process -Name xabuild -ErrorVariable xabuildError -PassThru
+    if ($xabuildError) { Write-Host "A process named 'xabuild' could not be found or stopped." }
+    Stop-Process -Name MSBuild -ErrorVariable msbuildError -PassThru
+    if ($msbuildError) { Write-Host "A process named 'MSBuild' could not be found or stopped." }
+    Stop-Process -Name adb -ErrorVariable adbError -PassThru
+    if ($adbError) { Write-Host "A process named 'adb' could not be found or stopped." }
   displayName: kill leftover processes
-  failOnStderr: false
+  ignoreLASTEXITCODE: true
```

Comment:
Since this list might grow someday, should we try this trick ([found here](https://stackoverflow.com/a/25558907)):
```powershell
# List all processes
Get-Process
# Get specific processes to kill (case-insensitive)
$processes = Get-Process | Where-Object {$_.ProcessName -match 'adb|msbuild|nunit3-console|xabuild' }
Write-Output $processes
$processes | Stop-Process
```
I don't know if we actually need to print if each one was running or not, since we have the `Get-Process` call at the top that lists everything. We need to keep that first call in case we find some new process hanging around.

Reply:
This particular syntax didn't work for me, but I made a similar change that should be more extensible in the future.